packages/backend/server/src/core/user/resolver.ts

```diff
@@ -231,4 +259,36 @@ export class UserManagementResolver {
     await this.user.deleteUser(id);
     return { success: true };
   }
+
+  @Mutation(() => UserType, {
+    description: 'Update a user',
+  })
+  async updateUser(
+    @Args('id') id: string,
+    @Args('input') input: ManageUserInput
+  ): Promise<UserType> {
+    const user = await this.db.user.findUnique({
+      select: { ...this.user.defaultUserSelect, password: true },
+      where: { id },
+    });
+
+    if (!user) {
+      throw new UserNotFound();
+    }
+    validators.assertValidEmail(input.email);
+    if (input.email !== user.email) {
+      const exists = await this.db.user.findFirst({
+        where: { email: input.email },
+      });
+      if (exists) {
+        throw new Error('Email already exists');
+      }
+    }
+    return sessionUser(
+      await this.user.updateUser(user.id, {
+        name: input.name ?? input.email.slice(0, input.email.indexOf('@')),
```

Comment:
do not use email prefix as name when updating, check input name existance and length